src/py/flwr/server/superlink/state/in_memory_state.py

```diff
@@ -44,11 +44,15 @@ def __init__(self) -> None:
         self.task_ins_store: Dict[UUID, TaskIns] = {}
         self.task_res_store: Dict[UUID, TaskRes] = {}
 
+        # Count undelivered task_ins/task_res
+        self._undelivered_task_ins_cnt = 0
+        self._undelivered_task_res_cnt = 0
+
         self.client_public_keys: Set[bytes] = set()
         self.server_public_key: Optional[bytes] = None
         self.server_private_key: Optional[bytes] = None
 
-        self.lock = threading.Lock()
+        self.lock = threading.RLock()
```

Comment:
Do we need to make it `RLock`?

Reply:
Maybe it's not necessary for now. I've changed it back to `Lock`.